src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -25,6 +25,7 @@
 #include "precompiled.hpp"
 #include "jvm.h"
 #include "jimage.hpp"
+#include "cds/cds_globals.hpp"
```

Comment:
Doesn't this, and similar, need to be guarded by INCLUDE_CDS?

Reply:
The motivation is modularity and avoid putting everything inside globals.hpp. There are already other files with the same style. E.g., share/gc/shared/tlab_globals.hpp

Some of the flags are accessed in contexts where CDS is not enabled, so for the time being I am leaving them always available.

If we decide to make some or all of these flags optional, the `#if INCLUDE_CDS` can be put inside cds_globals.hpp, instead of every file that uses it.

Reply:
Are you saying that none of the CDS flags are already actually gated on INCLUDE_CDS?

But good point about putting the guard inside the cds_globals.hpp file - obvious :)

Reply:
Unfortunately there are a few references to the CDS flags like SharedBaseAddress that are not gated.

This PR makes it easy to detect such cases (by adding an empty CDS_FLAGS macro in cds_globals.hpp if INCLUDE_CDS is false). I think such cleanup can be done in a follow-up PR.

Reply:
I filed https://bugs.openjdk.org/browse/JDK-8293135 to deal with such use of the CDS flags when CDS is disabled. There are potential compatibility issues, so we need some more discussion before proceeding.

Reply:
Okay. Thanks